R/watchdog.R

```diff
@@ -0,0 +1,128 @@
+
+# whether or not the user has enabled the renv watchdog in this session
+`_renv_watchdog_enabled` <- NULL
+
+# metadata related to the running watchdog process, if any
+`_renv_watchdog_metadata` <- NULL
+
+renv_watchdog_init <- function() {
+
+  `_renv_watchdog_enabled` <<- renv_watchdog_enabled_impl()
+  if (!`_renv_watchdog_enabled`)
+    return()
+
+  if (!renv_watchdog_running())
+    renv_watchdog_start()
+
+}
+
+renv_watchdog_enabled <- function() {
+  `_renv_watchdog_enabled`
+}
+
+renv_watchdog_enabled_impl <- function() {
+
+  # skip in older versions of R; we require newer APIs
+  if (getRversion() < "4.0.0")
+    return(FALSE)
+
+  # skip if disabled
+  enabled <- Sys.getenv("RENV_WATCHDOG_ENABLED", unset = "TRUE")
+  if (!truthy(enabled))
+    return(FALSE)
+
+  # skip during R CMD check (but not when running tests)
+  checking <- renv_envvar_exists("_R_CHECK_PACKAGE_NAME_")
```

Comment:
Use `renv_package_checking()`?

Reply:
That also checks if `is_testing()` so it doesn't seem like the right call (unless we want to pull that out)

https://github.com/rstudio/renv/blob/a04863eda8b8a871f09e8c51d541f56e691a2566/R/package.R#L405